src/ser/mod.rs

```diff
@@ -50,11 +50,12 @@ where W: Write {
     namespaces: Vec<(String, String)>,
     current_tag: String,
     current_tag_attrs: Option<HashMap<&'static str, String>>,
+    open_document: bool,
```

Comment:
I don't love this name or adding a one-off boolean.

An alternative I considered was a "Vec<XmlEvent>" for things to emit as an opening, and then have `new` call that with the default opening, but then I needed a bunch of lifetimes which seemed worse.

Another potential option would be have some way to specify that a struct is a processing instruction and then serialize it all as a list, like:
```
#[xml(opening)]
struct XmlOpening {
   version:  ...
   standalone: ...
   ...
}

#[xml(processing_instruction)]
struct OfxHeader {
   version: ...
   ofx_header: ...
   ...
}

let val = vec![
  XmlOpening{...},
  OfxHeader{...},
  Document{...}
]
```

but the way you have it where it emits a default opening probably covers 90% of use cases, so requiring people to always specify their opening seems a little heavy handed.

Reply:
Thanks for your support and for sharing this use case and PR.

Looking at the documentation of `xml-rs` again, I notice that the default behaviour is to output a default XML declaration. Thus, it is not strictly required to emit a StartDocument event as long as the default configuration is used.

By removing the line of code that emits the StartDocument event (https://github.com/punkstarman/serde-xml/blob/master/src/ser/mod.rs#L98) and also removing any dead code, the test you added passes. All the other serialization tests fail because the default XML declaration writes the encoding in lowercase instead of uppercase (`utf-8` instead of `UTF-8`). A quick search-replace fixes this.

I have pushed the resulting code for comparison: https://github.com/punkstarman/serde-xml/tree/custom-header.

What do you think?

Reply:
That looks great, and much cleaner!

I think your solution is strictly better than the one I proposed. As long as `new_from_writer` stays `pub`, my use case is totally covered.

Reply:
Sounds good. I've pushed my commit onto your branch and I am ready to accept if you're OK.